packages/@aws-cdk/aws-sns/lib/policy.ts

```diff
@@ -21,19 +26,15 @@ export class TopicPolicy extends Resource {
   /**
    * The IAM policy document for this policy.
    */
-  public readonly document = new PolicyDocument({
-    // statements must be unique, so we use the statement index.
-    // potantially SIDs can change as a result of order change, but this should
-    // not have an impact on the policy evaluation.
-    // https://docs.aws.amazon.com/sns/latest/dg/AccessPolicyLanguage_SpecialInfo.html
-    assignSids: true,
-  });
+  public readonly document: PolicyDocument;
 
   constructor(scope: Construct, id: string, props: TopicPolicyProps) {
     super(scope, id);
 
+    this.document = props.policyDocument;
```

Comment:
We can change this to `this.document = props.policyDocument ?? this.document` so that it uses the default value of the property from